packages/workflow-core/src/lib/plugins/common-plugin/types.ts

```diff
@@ -0,0 +1,31 @@
+import { TContext, Transformers } from "../../utils";
+import { SerializableValidatableTransformer } from "../external-plugin";
+import { ChildPluginCallbackOutput } from "../../types";
+
+export interface ISerializableCommonPluginParams
+  extends Omit<IterativePluginParams, 'action' | 'iterateOn'> {
+  iterateOn: Omit<SerializableValidatableTransformer, 'schema'>['transform'];
+  response: SerializableValidatableTransformer;
+  actionPluginName: string;
+
+  invoke?(...args: Array<any>): any;
+}
+
+export interface IterativePluginParams {
+  name: string;
+  stateNames: Array<string>;
+  iterateOn: Transformers;
+  action: (context: TContext) => Promise<any>
+  successAction?: string;
+  errorAction?: string;
+}
+
+export interface ChildWorkflowPluginParams {
+  name: string;
+  parentWorkflowRuntimeId: string;
+  definitionId: string;
+  stateNames?: Array<string>;
+  transformers?: Transformers;
+  initEvent?: string;
+  action: (childCallbackInput: ChildPluginCallbackOutput) => Promise<void>
+}
```

Comment:
*non-blocker - should use a base interface with name, stateNames, and maybe action.

Reply:
will happen in the future after we finalize more plugins